Modules/Relationships/src/Relationships.ConsumerApi/Controllers/RelationshipsController.cs

```diff
@@ -112,6 +116,9 @@ public async Task<IActionResult> GetChangeById(RelationshipChangeId id, Cancella
     [ProducesResponseType(typeof(HttpResponseEnvelopeResult<CreateRelationshipResponse>), StatusCodes.Status200OK)]
     public async Task<IActionResult> CreateRelationship(CreateRelationshipCommand request, CancellationToken cancellationToken)
     {
+        var relationshipTemplate = await _mediator.Send(new GetRelationshipTemplateQuery { Id = request.RelationshipTemplateId }, cancellationToken);
```

Comment:
Why didn't you move this call to `EnsurePeerIsNotToBeDeleted` as well?

Reply:
In this instance, we have a Relationship Template Id, which we use to retrieve the template itself and the identity address.
In other cases, we have a Relationship Id, and we retrieve an actual Relationship.